compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -544,9 +546,15 @@ checkValidProjectionsKey = \case
   expr ->
     throwWithContext (EInvalidKeyExpression expr)
 
-checkTemplateKey :: MonadGamma m => ExprVarName -> Qualified TypeConName -> Maybe TemplateKey -> m ()
-checkTemplateKey param tcon mbKey =
-  for_ mbKey $ \TemplateKey{..} -> do
+checkSupportsContractKeys :: MonadGamma m => m ()
+checkSupportsContractKeys = do
+    v <- view lfVersion
+    unless (supportsContractKeys v) $
+        throwWithContext $ EUnsupportedFeature "Contract keys" version1_3
```

Comment:
Not such a fan that `supportsContractKeys` is a nicely abstracted predicate, and then the error message repeats the version miles away from it

Reply:
I totally agree. I'll clean that up in a followup PR for all features that depend on the DAML-LF version. Same for the other remark.